aiinfra-cluster/variables.tf

```diff
@@ -14,79 +14,78 @@
   * limitations under the License.
   */
 
-variable "name_prefix" {
-  description = ""
+variable "project_id" {
+  description = "The project_id to create the resources for GPU cluster."
   type        = string
 }
 
-variable "gpu_per_vm" {
-  description = ""
-  type        = number
+variable "service_account" {
+  description = "Service account to attach to the instance. See https://www.terraform.io/docs/providers/google/r/compute_instance_template.html#service_account."
+  type = object({
+    email  = string,
+    scopes = set(string)
+  })
```

Comment:
[`_env_var_util.sh:59`](https://github.com/GoogleCloudPlatform/ai-infra-cluster-provisioning/blob/develop/scripts/_env_var_util.sh#L59) still sets `service_account` to a `string` (instead of `{ email: string, scopes: [string] }` like it is changed to here). @soumyapani @DmitryKakurin it seems a little unintuitive (and seems to be a bit less maintainable) to have a default value both here and in that script. Should this default `scopes` just get moved to that script since the default email there is generated using that `gcloud projects describe` call on [line 54](https://github.com/GoogleCloudPlatform/ai-infra-cluster-provisioning/blob/develop/scripts/_env_var_util.sh#L54)?

Reply:
good catch @stevenBorisko.
I recently found out that is is a bug and fails the startup commands to run successfully.
I am sending out a fix shortly.
Lets discuss about a better testing story. I think the current testing that we have does not cover all the cases.